src/CMakeLists.txt

```diff
@@ -15,7 +15,14 @@
 # specific language governing permissions and limitations
 # under the License.
 
-cmake_minimum_required(VERSION 3.5.2)
+cmake_minimum_required(VERSION 3.11.0)
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    # require at least gcc 5.4.0
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.4.0)
```

Comment:
you guys can be more aggressive to allow only gcc7 actually, since you have bundled gcc already.